src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -366,6 +366,20 @@ export default class QueryEditorUI extends Component<Props, State> {
         editor.setValue(`\n`.repeat(10));
         this.setState({ lineCount: editor.getModel()?.getLineCount() });
         this.inputRef = editor;
+        const handleEnterPress = () => {
+          this.onSubmit(this.props.query);
+        };
+
+        const disposable = editor.onKeyDown((e) => {
+          if (e.metaKey && e.keyCode === monaco.KeyCode.Enter) {
+            e.preventDefault();
+            handleEnterPress();
+          }
+        });
```

Comment:
```suggestion
        const disposable = editor.onKeyDown((e) => {
          if (e.metaKey && e.keyCode === monaco.KeyCode.Enter) {
            e.preventDefault();
            this.onSubmit(this.props.query);
          }
        });
```
Nit: handleEnterPress doesnt seem to add any value here. What do you think?